Sources/TuistKit/Generator/GeneratorFactory.swift

```diff
@@ -14,13 +14,15 @@ protocol GeneratorFactorying {
     /// - Parameter config: The project configuration.
     /// - Parameter sources: The list of targets whose sources should be inclued.
     /// - Parameter xcframeworks: Whether targets should be cached as xcframeworks.
+    /// - Parameter xcframeworksType: Output type of xcframeworks to be cached (device/simulator).
     /// - Parameter cacheProfile: The caching profile.
     /// - Parameter ignoreCache: True to not include binaries from the cache.
     /// - Returns: The generator for focused projects.
     func focus(
         config: Config,
         sources: Set<String>,
         xcframeworks: Bool,
+        xcframeworksType: CacheXCFrameworkType?,
```

Comment:
Can we just get the CacheOutputType here?
We should avoid invalid combinations (e.g. xcframeworks true and xcframeworksType non nil), and this should be done in the command arguments validation

Reply:
Same in all other places we are passing both